sstables/mx/reader.cc

```diff
@@ -662,7 +662,10 @@ class mp_row_consumer_m {
 
 // data_consume_rows_context_m remembers the context that an ongoing
 // data_consume_rows() future is in for SSTable in 3_x format.
-class data_consume_rows_context_m : public data_consumer::continuous_data_consumer<data_consume_rows_context_m> {
+template <typename Consumer>
```

Comment:
Please add (or find) a concept for this. Otherwise it's very hard for the uninitiated (or even the initiated) to reverse-engineer the requirements from Consumer.